com.unity.multiplayer.mlapi/Tests/Runtime/NetworkVariableTests.cs

```diff
@@ -1,26 +1,143 @@
+
+        private const uint k_TestUInt = 0xdeadbeef;
+
+        private const int k_TestVal1 = 111;
+        private const int k_TestVal2 = 222;
+        private const int k_TestVal3 = 333;
+
+        private const int k_TestKey1 = 0x0f0f;
+        private const int k_TestKey2 = 0xf0f0;
+
+        private NetworkVariableTest m_ServerComp;
+        private NetworkVariableTest m_ClientComp;
+
+        private readonly bool m_TestWithHost = false;
+
+        [UnitySetUp]
+        public override IEnumerator Setup()
         {
-            // Create, instantiate, and host
-            Assert.IsTrue(NetworkManagerHelper.StartNetworkManager(out _));
+            yield return StartSomeClientsAndServerWithPlayers(useHost: m_TestWithHost, nbClients: NbClients,
+                updatePlayerPrefab: playerPrefab =>
+                {
+                    var networkTransform = playerPrefab.AddComponent<NetworkVariableTest>();
+                });
+
+            // This is the *SERVER VERSION* of the *CLIENT PLAYER*
+            var serverClientPlayerResult = new MultiInstanceHelpers.CoroutineResultWrapper<NetworkObject>();
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.GetNetworkObjectByRepresentation(
+                x => x.IsPlayerObject && x.OwnerClientId == m_ClientNetworkManagers[0].LocalClientId,
+                m_ServerNetworkManager, serverClientPlayerResult));
+
+            // This is the *CLIENT VERSION* of the *CLIENT PLAYER*
+            var clientClientPlayerResult = new MultiInstanceHelpers.CoroutineResultWrapper<NetworkObject>();
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.GetNetworkObjectByRepresentation(
+                x => x.IsPlayerObject && x.OwnerClientId == m_ClientNetworkManagers[0].LocalClientId,
+                m_ClientNetworkManagers[0], clientClientPlayerResult));
+
+            var serverSideClientPlayer = serverClientPlayerResult.Result;
+            var clientSideClientPlayer = clientClientPlayerResult.Result;
+
+            m_ServerComp = serverSideClientPlayer.GetComponent<NetworkVariableTest>();
+            m_ClientComp = clientSideClientPlayer.GetComponent<NetworkVariableTest>();
+
+            m_ServerComp.TheList.Clear();
+
+            if (m_ServerComp.TheList.Count > 0)
+            {
+                throw new Exception("server network list not empty at start");
+            }
+            if (m_ClientComp.TheList.Count > 0)
+            {
+                throw new Exception("client network list not empty at start");
+            }
         }
 
         /// <summary>
         /// Runs generalized tests on all predefined NetworkVariable types
         /// </summary>
         [UnityTest]
-        public IEnumerator TestAllNetworkVariableTypes()
+        public IEnumerator AllNetworkVariableTypes()
         {
+            // Create, instantiate, and host
+            Assert.IsTrue(NetworkManagerHelper.StartNetworkManager(out _));
```

Comment:
@NoelStephensUnity I did this on purpose vs. being in setup.  Maybe it doesn't hurt for this also to be in setup, but I'm not exactly sure how our setups play together

Reply:
So, the setup ([UnitySetUp]) executes before each unit test and the Teardown ([UnityTearDown]) executes at the end of the unit test.  Instantiating NetworkManager in the unit test method and destroying it at the end of the unit test method can become problematic if the test fails as it will break the call stack and leave the instance of the NetworkManager.  Using the setup and teardown as a place to instantiate and destroy things assures that even upon failure it will clean itself up so the unit tests after it will not have class artifacts sitting around from the previous failed unit test that could potentially adversely impact the remaining unit tests.  This helps narrow down where the failure is specifically as the left behind artifacts might not impact all of the remaining unit tests making it seem like a more "global breakage" as opposed to perhaps a single unit test localized breakage.